plugins/by-name/tinygit/default.nix

```diff
@@ -0,0 +1,62 @@
+{
+  lib,
+  pkgs,
+  ...
+}:
+lib.nixvim.plugins.mkNeovimPlugin {
+  name = "tinygit";
+  packPathName = "nvim-tinygit";
+  package = "nvim-tinygit";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  extraOptions = {
+    curlPackage = lib.mkPackageOption pkgs "curl" {
+      nullable = true;
+    };
+
+    gitPackage = lib.mkPackageOption pkgs "git" {
+      nullable = true;
+    };
```

Comment:
Wonder if we should create top level options for some of these common dependencies.